--- lldp_util.c	2016-03-31 16:44:11.000000000 +0000
+++ ../lldp_util.c	2016-03-31 18:41:41.336813782 +0000
@@ -1197,7 +1197,7 @@
 int get_arg_val_list(char *ibuf, int ilen, int *ioff,
 			    char **args, char **argvals)
 {
-	u8 arglen;
+	u8 arglen = 0;
 	u16 argvalue_len;
 	int *arglens = NULL;
 	int *argvallens = NULL;
@@ -1265,7 +1265,7 @@
 
 int get_arg_list(char *ibuf, int ilen, int *ioff, char **args)
 {
-	u8 arglen;
+	u8 arglen = 0;
 	int *arglens = NULL;
 	int *p;
 	int numargs;
